content/challenges/medium/4 - Reverse Words/func.js

```diff
@@ -11,5 +11,36 @@
  * @returns {string} The string with the proper words reversed
  */
 module.exports = (str) => {
-    // your code here
+    let result = '';
+    let currentWord = '';
+
+    for (let i = 0; i < str.length; i++) {
+        if (str[i] !== ' ') {
+            currentWord += str[i];
+        } else {
+            if (currentWord.length >= 5) {
+                let flippedWord = '';
+                for (let k = currentWord.length - 1; k >= 0; k--) {
+                    flippedWord += currentWord[k];
+                }
+                result += flippedWord + ' ';
+            } else {
+                result += currentWord + ' ';
+            }
+
+            currentWord = '';
+        }
+    }
+
+    if (currentWord.length >= 5) {
+        let flippedWord = '';
+        for (let k = currentWord.length - 1; k >= 0; k--) {
+            flippedWord += currentWord[k];
+        }
+        result += flippedWord;
+    } else {
+        result += currentWord;
+    }
+
+    return result;
```

Comment:
this probably works. And I'm happy you used all the primitive string manipulation ways to make this work.....now... do it again but use someString.split(' ') and see just how much easier it is :)

<img width="417" alt="image" src="https://github.com/user-attachments/assets/c6114ba0-4b5b-43aa-bdc7-6280556f049c" />